RHELPLAN-41572 - Logging - Convert each subrole to a pair of task/var files

nhosoi opened the pull request:
Apply the idea of "Convert the files output subrole to a pair of task/var files" by @pcahyna to the rest of the subroles.

```
commit 4b77713745569589ef966c13825495c7846a9cae
Author: Pavel Cahyna <pcahyna@redhat.com>
Date:   Fri Apr 24 23:49:38 2020 +0200

    Convert the files output subrole to a pair of task/var files
    
    UNTESTED
    
    Remove some debug stuff in the task file
```

nhosoi committed:
RHELPLAN-32351 	Add the flow control between inputs and outputs

- Introducing logging_flows to specify the inputs and outputs relationships.
  The relationship could be M inputs to N outputs.
  The configuration looks like this.  This example shows log messages from inputX,
  Y and Z are sent to outputA; log messages from inputX and Y are sent to outputB.

    logging_outputs:
      - name: outputA
        type: otypeA
        <<extra_params>>
      - name: outputB
        type: otypeB
        <<extra_params>>
    logging_inputs:
      - name: inputX
        type: itypeX
        <<extra_params>>
      - name: inputY
        type: itypeY
        <<extra_params>>
      - name: inputZ
        type: itypeZ
        <<extra_params>>
    logging_flows:
      - name: flow0
        inputs: [outputA, outputB]
        outputs: [inputX, inputY]
      - name: flow1
        inputs: [outputA]
        outputs: [inputZ]

- It is implemented with the rsyslog ruleset.  The rsyslog configuration is placed in ruleset
  in the output config file.  And input config "call TARGET_RULESET_NAME" base upon the
  logging_flows values.
- Note: logging_inputs used to be logs_collections.  It was renamed to adjust to logging_outputs.

- Updated Docs:
  README.md, roles/rsyslog/README.md and design_docs/rsyslog_new_input_output_roles_proposal.md

- New CI tests:
  tests_files_elasticsearch.yml, tests_combination.yml, tests_combination2.yml, tests_files_forwards_implicit.yml

- ansible-lint issue
  Workaround this warning:
  [WARNING]: Unable to convert data using to_nice_json, falling back to to_json: 'item' is undefined

pcahyna committed:
Convert the files output subrole to a pair of task/var files

UNTESTED

Remove some debug stuff in the task file

nhosoi committed:
RHELPLAN-41572 - Logging - Convert each subrole to a pair of task/var files

Apply the idea of "Convert the files output subrole to a pair of task/var files" by @pcahyna to the rest of the subroles.

richm commented:
lgtm

nhosoi commented:
Thanks for reviewing the pr, @richm.  I'm updating the doc now...  When submitting a PR in linux-system-roles/logging (once pr100 is merged), I'm appending the doc commit, too.  Thanks!

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}_cleanup.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
```
what is this variable used for?

nhosoi replied:
Thanks!  I missed to add this condition...
`- tasks_abspath is exists`

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}_cleanup.yml"
```
I would create a subdirectory for every former input subrole, so make `inputs/{{ input_item.type }}_cleanup.yml` into `inputs/{{ input_item.type }}/cleanup.yml` (and `inputs/{{ input_item.type }}.yml` into `inputs/{{ input_item.type }}/main.yml`)

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
+      when:
+        - rsyslog_enabled | bool
+        - output_item | d([])
+        - varfile_abspath is exists
```
is it needed? Are there outputs which do not have a variable file?

nhosoi replied:
A good question.  Probably, not. :)  Let me remove the line.

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
       when:
         - rsyslog_enabled | bool
         - output_item | d([])
+        - tasks_abspath is exists
```
is it needed? Are there outputs which do not have a task file?

nhosoi replied:
Ok...  So, you are suggesting to get rid of {tasks,varfile}_abspath completely...

pcahyna replied:
yes, if there ever is an input or output which lacks a var file or task file, it is probably simpler to add a dummy empty one.

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
```
is it needed? Are there inputs which do not have a task file?

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
+      when:
+        - rsyslog_enabled | bool
+        - input_item | d([])
+        - tasks_abspath is exists
```
is it needed? Are there inputs which do not have a task file?

pcahyna commented on roles/rsyslog/tasks/main.yml:
```diff
@@ -130,37 +130,80 @@
         - rsyslog_enabled | bool
         - input_item | d([])
-        - input_item.state | d('present') != 'absent'
+        - varfile_abspath is exists
```
is it needed? Are there inputs which do not have a vars file?

nhosoi committed:
- Moving roles/rsyslog/{tasks,vars}/{inputs,outputs}/type.yml to
  roles/rsyslog/{tasks,vars}/{inputs,outputs}/type/main.yml.
- Eliminating generating task_abspath and varfile_abspath and checking
  the existence of the files.
- Updating README.md and doc files in design_docs.
